diff --git a/3rdparty/filament/filament_build.cmake b/3rdparty/filament/filament_build.cmake
index 16c590b..3726c4d 100644
--- a/3rdparty/filament/filament_build.cmake
+++ b/3rdparty/filament/filament_build.cmake
@@ -107,6 +107,12 @@ if(MSVC)
         BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config ${FILAMENT_BUILD_CONFIG})
 endif()
 
+if(FILAMENT_PATCH_FILE)
+    set(FILAMENT_PATCH_COMMAND COMMAND ${CMAKE_COMMAND}
+        -DPATCH_FILE=${FILAMENT_PATCH_FILE} -DSOURCE_DIR=<SOURCE_DIR>
+        -P ${Open3D_SOURCE_DIR}/cmake/apply_patch.cmake)
+endif()
+
 ExternalProject_Add(
     ext_filament
     PREFIX filament
@@ -115,6 +121,7 @@ ExternalProject_Add(
     DOWNLOAD_DIR "${OPEN3D_THIRD_PARTY_DOWNLOAD_DIR}/filament"
     # 0001: Implements VulkanDriver::importTextureR for zero-copy 3DGS texture sharing.
     PATCH_COMMAND ${CMAKE_COMMAND} -DPATCH_FILE=${Open3D_3RDPARTY_DIR}/filament/patches/0001-importTextureR.patch -DSOURCE_DIR=<SOURCE_DIR> -P ${Open3D_SOURCE_DIR}/cmake/apply_patch.cmake
+    ${FILAMENT_PATCH_COMMAND}
     UPDATE_COMMAND ""
     CMAKE_ARGS
         ${ExternalProject_CMAKE_ARGS}
@@ -137,6 +144,7 @@ ExternalProject_Add(
         -DUSE_STATIC_LIBCXX=ON
         -DFILAMENT_SKIP_SDL2=ON
         -DFILAMENT_SKIP_SAMPLES=ON
+        -DFILAMENT_SUPPORTS_EGL_ON_LINUX=ON
         -DFILAMENT_OPENGL_HANDLE_ARENA_SIZE_IN_MB=20 # to support many small entities
         -DSPIRV_WERROR=OFF
         ${filament_build_command}
